includes/wikia/services/ImagesService.class.php

```diff
@@ -157,27 +157,41 @@ private static function isDataTagImage($url) {
 	 */
 	public static function getThumbUrlFromFileUrl($imageUrl, $destSize, $newExtension = null) {
 		if (!empty($imageUrl)) {
-			if ( !self::IsExternalThumbnailUrl($imageUrl) ) {
-				$imageUrl = str_replace('/images/', '/images/thumb/', $imageUrl);
-			} else {
-				$imageUrl = $imageUrl;
-			}
+			if ( VignetteRequest::isVignetteUrl( $imageUrl ) ) {
+				$generator = VignetteRequest::fromUrl( $imageUrl, true );
 
+				if ( strpos( $destSize, "px" ) !== false ) {
+					list( $width, $_ ) = explode( "px", $destSize );
+					$generator->scaleToWidth( $width );
+				} else {
+					list( $width, $height ) = explode( "x", $destSize );
+					$generator
+						->fixedAspectRatio()
+						->height( $height )
+						->width( $width );
+				}
```

Comment:
This is getting dicey with the backwards compatibility of the `$destSize` param. It accepts things like `200px` or `200x200`. I would favor breaking the vignette block out into a separate function that handles this behavior in a more sane way. Something like:

```php
public static function getThumbUrlFromFileUrl($imageUrl, $destSize, $newExtension = null) {
...
   if ( VignetteRequest::isVignetteUrl( $imageUrl ) ) {
      list($width, $height) = self::parseGetThumbUrlFromFileUrlDestSize($destSize);
      $url = self::getThumbUrlFromVignetteUrl($imageUrl, $width, $height);
   }
}
```

This way, the logic should be easier to follow, test, and allow deprecation and removal of the old stuff.